pkg/eventhandler/eventhandler.go

```diff
@@ -82,6 +82,8 @@ func (n *EventHandler) HandleGrantEvent(ctx context.Context, log *zap.SugaredLog
 		}
 		requestEvent = access.NewGrantFailedEvent(gq.Result.ID, event.Time, oldStatus, newStatus, grantFailedEvent.Reason)
 		log.Infow("inserting request event for grant failed")
+		//if failed then we want to update the end time of the grant to be completed
+		gq.Result.Grant.End = event.Time
```

Comment:
This is not the right way to do this, the grant start and end time should not be modified after it is created.
These fields indicate the requested grant.

Reply:
The specific change that needs to be made is on the DDBKeys method of request.
There is a caculated field there which is used to sort the request into either upcoming or past